### ISSUE TITLE
API pod: `minikube.net` in StaticClients RedirectURIs ?

### ISSUE THREAD

**Issue by genedev22** [AzBuilder/terrakube-helm-chart#147]
Greetings.

I'm trying to deploy chart version 3.21.4, but I'm seeing references to http://terrakube-ui.minikube.net that are not being overwritten?
1. It shows up as an error when trying to login from the UI:
```
Bad Request
Unregistered redirect_uri ("http://terrakube-ui.minikube.net").
```
2. I see a reference to http://terrakube-ui.minikube.net in the API pod logs:
```
$ k logs terrakube-api-577dfbd5b8-bm9x2 | grep -i minikube -C 5
2024-10-06T01:20:06.052Z  INFO 1 --- [           main] o.t.a.p.scheduler.module.CacheService    : jobDetail is null false
2024-10-06T01:20:06.052Z  INFO 1 --- [           main] o.t.a.p.scheduler.module.CacheService    : Delete Old Quartz Job
2024-10-06T01:20:06.355Z  INFO 1 --- [           main] o.t.a.p.scheduler.module.CacheService    : Reschedule with new frequency
2024-10-06T01:20:06.355Z  INFO 1 --- [           main] o.t.a.p.scheduler.module.CacheService    : Create Schedule Job Trigger DEFAULT.TerrakubeV2_ModuleRefresh
2024-10-06T01:20:08.897Z  INFO 1 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 1 endpoint beneath base path '/actuator'
2024-10-06T01:20:09.132Z  INFO 1 --- [           main] o.t.a.p.s.a.dex.DexWebSecurityAdapter    : Loading CORS http://terrakube-ui.minikube.net
2024-10-06T01:20:19.738Z  INFO 1 --- [           main] com.yahoo.elide.Elide                    : Registering serde for type : class java.time.Instant
2024-10-06T01:20:19.812Z  INFO 1 --- [           main] com.yahoo.elide.Elide                    : Registering serde for type : class java.time.OffsetDateTime
2024-10-06T01:20:19.812Z  INFO 1 --- [           main] com.yahoo.elide.Elide                    : Registering serde for type : class java.util.TimeZone
2024-10-06T01:20:19.813Z  INFO 1 --- [           main] com.yahoo.elide.Elide                    : Registering serde for type : class java.net.URL
2024-10-06T01:20:19.813Z  INFO 1 --- [           main] com.yahoo.elide.Elide                    : Registering serde for type : class java.util.Date
```
3. There is a similar log message in the Registry pod logs.

This is the `values.yaml` that I am using. In case it is relevant, I am using a separate ingress for an AWS ALB load balancer controller, which I'll also show. 

```yaml
## Global Name
name: "terrakube"

global:
  imagePullSecrets: []

security:
  useOpenLDAP: false
  adminGroup: "TERRAKUBE_ADMIN"
  patSecret: "<snip>"
  internalSecret: "<snip>"
  dexClientId: cognito-app
  dexClientScope: "email openid profile offline_access groups"
  dexIssuerUri: "https://terrakube-api.<snip>.com/dex"
  existingSecret: false

## Dex
dex:
  enabled: true
  existingSecret: false
  config:
    issuer: "https://terrakube-api.<snip>.com/dex"

    storage:
      type: memory
    web:
      http: 0.0.0.0:5556
      allowedOrigins: ['*']
      skipApprovalScreen: true
    oauth2:
      responseTypes: ["code", "token", "id_token"]

    connectors:
    - type: oidc
      id: cognito
      name: AWS Cognito
      config:
        issuer: "https://cognito-idp.us-east-1.amazonaws.com/us-east-1_<snip>"
        clientID: "<snip>"
        clientSecret: "<snip>"
        redirectURI: "https://terrakube-api.<snip>.com/dex/callback"
        scopes:
          - openid
          - email
          - profile
        insecureSkipEmailVerified: true
        insecureEnableGroups: true
        userNameKey: "cognito:username"
        claimMapping:
          groups: "cognito:groups"
 

    staticClients:
    - id: cognito-app
      redirectURIs:
      - 'https://terrakube-ui.<snip>.com' # This is not loading in the Dex redirect.
      - '/device/callback'
      - 'http://localhost:10000/login'
      - 'http://localhost:10001/login'
      name: 'cognito-app'
      public: true

## Terraform Storage
storage:
  defaultStorage: false
  aws:
    accessKey: "<snip>"
    secretKey: "<snip>"
    bucketName: "<snip>"
    region: "us-east-1"

# Default Redis Configuration
redis:
  architecture: "standalone"
  auth:
    enabled: true
    password: "<snip>"

## API properties
api:
  existingSecret: false
  enabled: true
  image: azbuilder/api-server
  version: ""
  replicaCount: "1"
  serviceType: "ClusterIP"
  serviceAccountName: ""
  rbac:
    create: false
    roleName: "terrakube-api-role"
    roleBindingName: "terrakube-api-role-binding"
  secrets:
   - terrakube-api-secrets
  resources: {}
  podLabels: {}
  defaultDatabase: false
  defaultRedis: true
  loadSampleData: false
  terraformReleasesUrl: "https://releases.hashicorp.com/terraform/index.json"
  securityContext: {}
  containerSecurityContext: {}
  imagePullSecrets: []
  initContainers: []
  cache:
    moduleCacheMaxTotal: "128"
    moduleCacheMaxIdle: "128"
    moduleCacheMinIdle: "64"
    moduleCacheTimeout: "600000"
    moduleCacheSchedule: "0 */3 * ? * *"
  properties:
    databaseType: "POSTGRESQL"
    databaseHostname: "<snip>"
    databaseName: "terrakube"
    databaseUser: "terrakube"
    databaseSchema: "public"
    databasePassword: "<snip>"
    databaseSslMode: "require"
    databasePort: "5432"
    redisHostname: ""
    redisPassword: ""

## Executor properties
executor:
  existingSecret: false
  enabled: true
  image: "azbuilder/executor"
  version: ""
  replicaCount: "1"
  serviceType: "ClusterIP"
  serviceAccountName: ""
  secrets:
   - terrakube-executor-secrets
  resources: {}
  podLabels: {}
  podAnnotations: {}
  apiServiceUrl: "http://terrakube-api-service:8080"
  properties:
    toolsRepository: "https://github.com/AzBuilder/terrakube-extensions"
    toolsBranch: "main"
  securityContext: {}
  containerSecurityContext: {}
  imagePullSecrets: []
  initContainers: []

## Registry properties
registry:
  enabled: true
  existingSecret: false
  image: azbuilder/open-registry
  version: ""
  replicaCount: "1"
  serviceType: "ClusterIP"
  serviceAccountName: ""
  secrets:
   - terrakube-registry-secrets
  resources: {}
  podLabels: {}
  securityContext: {}
  containerSecurityContext: {}
  imagePullSecrets: []
  initContainers: []

## UI Properties
ui:
  enabled: true
  existingSecret: false
  image: azbuilder/terrakube-ui
  version: ""
  replicaCount: "1"
  serviceType: "ClusterIP"
  serviceAccountName: ""
  resources: {}
  podLabels: {}
  securityContext: {}
  containerSecurityContext: {}
  imagePullSecrets: []
  initContainers: []

## Ingress properties
ingress:
  useTls: false
  includeTlsHosts: false
  ui:
    enabled: false
  api:
    enabled: false
  registry:
    enabled: false
  dex:
    enabled: false
```

Separate ALB ingress definition. I talked about this in Issue https://github.com/AzBuilder/terrakube-helm-chart/issues/93 as well.

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: alb-ingress
  namespace: terrakube
  annotations:
    alb.ingress.kubernetes.io/load-balancer-name: terrakube-alb
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS":443}]'
    alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:us-east-1:<account id>:certificate/<id of your SSL cert in ACM>
    alb.ingress.kubernetes.io/ssl-redirect: '443'
    alb.ingress.kubernetes.io/actions.ssl-redirect: '{"Type": "redirect", "RedirectConfig": { "Protocol": "HTTPS", "Port": "443", "StatusCode": "HTTP_301"}}'
    alb.ingress.kubernetes.io/group.name: alb-group
    alb.ingress.kubernetes.io/target-type: ip
spec:
  ingressClassName: alb
  rules:
    - host: terrakube-ui.<snip>.com
      http:
        paths:
          - path: "/*"
            pathType: ImplementationSpecific
            backend:
              service:
                name: terrakube-ui-service
                port:
                  number: 8080
    - host: terrakube-api.<snip>.com
      http:
        paths:
          - path: "/dex/*"
            pathType: ImplementationSpecific
            backend:
              service:
                name: terrakube-dex
                port:
                  number: 5556
          - path: "/*"
            pathType: ImplementationSpecific
            backend:
              service:
                name: terrakube-api-service
                port:
                  number: 8080
    - host: terrakube-reg.<snip>.com
      http:
        paths:
          - path: "/*"
            pathType: ImplementationSpecific
            backend:
              service:
                name: terrakube-registry-service
                port:
                  number: 8075
```

Thanks in advance.

**Comment by alfespa17** [comment]
Hello @genedev22 even if you disable the ingress to still need to overwrite the default values for the api, registry for example check all the default chart values in this lines, you will see there all the minikube references 

https://github.com/AzBuilder/terrakube-helm-chart/blob/efd3585024cd7236cabfd389ed090d3d2add721f/charts/terrakube/values.yaml#L279

**Comment by alfespa17** [comment]
Take a look at this example, even if you dísable the ingress config the api secret still need to know what will be the value for the domain

https://github.com/AzBuilder/terrakube-helm-chart/blob/efd3585024cd7236cabfd389ed090d3d2add721f/charts/terrakube/templates/secrets-api.yaml#L15

**Comment by alfespa17** [comment]
Another example for the registry secret cam be found here

https://github.com/AzBuilder/terrakube-helm-chart/blob/efd3585024cd7236cabfd389ed090d3d2add721f/charts/terrakube/templates/secrets-registry.yaml#L8

**Comment by genedev22** [comment]
@alfespa17 

Thanks. I was puzzled for a while about minikube.net references.
- Specifying the domains in the `values.yaml` Ingress helped and I was no longer seeing minikube.net references in the pod logs.
- I also set `useTls: true` and `includeTlsHosts: true` to accomodate the `dex.staticclients.redirectURIs` configuration.
- Yet for a while I kept getting the same error from my browser.

I kept getting the minikube.net error even despite removing minikube.net references from the Helm chart defaults and running `helm delete terrakube` followed by `helm install ...`. Eventually the error went away.

As to _why_ I kept seeing references to minikube.net: I'm guessing it was a browser caching issue. To help confirm the cluster wasn't using references to minikube.net:
1. I re-checked the pod logs.
2. Ran a recursive `grep` to find "minikube" in each pod.
3. Checked the environment variables via `env` in each pod.

Anyway, everything looks clear now. 

fyi: I added some more notes in Issue #93 where I found a combination of settings that allows me to log into the application.
